app/controllers/users_controller.rb

```diff
@@ -54,9 +54,24 @@ def update
     end
   end
 
+  def enable
+    user = User.find(params[:user_id])
+    user.update(:enabled => params[:enabled])
+    if user[:enabled] = true
+      flash[:notice] = "#{user.name} is now enabled"
+    else
+      flash[:notice] = "#{user.name} is now disabled"
+    end
+    redirect_to "/merchants"
+  end
+
   private
 
   def user_params
-    params.require(:user).permit(:name, :street, :city, :state, :zip, :email, :password)
+    if current_admin?
+      params.fetch(:user, {}).permit(:name, :street, :city, :state, :zip, :email, :password, :enabled, :user_id)
```

Comment:
This is interesting, I'm not familiar with fetch! Note to ask you to explain how this works.

Reply:
it was removed as it wound up being un-needed but could come in handy in the future. It was needed while i was going down a different path it enables both current user and the user object we want to edit as being permits.